cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1765,7 +1768,23 @@ table_with_metadata reader::impl::read(size_type skip_rows,
       // create the final output cudf columns
       for (size_t i = 0; i < _output_columns.size(); ++i) {
         column_name_info& col_name = out_metadata.schema_info.emplace_back("");
-        out_columns.emplace_back(make_column(_output_columns[i], &col_name, _stream, _mr));
+        auto col                   = make_column(_output_columns[i], &col_name, _stream, _mr);
+        if (_output_columns[i].type.id() == type_id::STRING &&
+            _force_binary_columns_as_strings.has_value() &&
+            !_force_binary_columns_as_strings.value()[i]) {
+          auto const& schema = _metadata->get_schema(_output_column_schemas[i]);
+          if (schema.converted_type == parquet::UNKNOWN) {
+            auto const num_rows = col->size();
+            auto data           = col->release();
+            col =
+              make_lists_column(num_rows,
+                                std::move(data.children[strings_column_view::offsets_column_index]),
+                                std::move(data.children[strings_column_view::chars_column_index]),
+                                UNKNOWN_NULL_COUNT,
+                                std::move(*data.null_mask));
+          }
+        }
+        out_columns.emplace_back(std::move(col));
```

Comment:
I feel like wrapping this whole process (make_column + this postprocess step) in a lambda or something would make things cleaner.   The fact that we have a global function called `make_column` is kind of ugly too, though changing that would touch a lot of things.